src/java/org/apache/cassandra/io/util/PathUtils.java

```diff
@@ -334,6 +336,52 @@ private static void deleteRecursive(Path path, RateLimiter rateLimiter, Consumer
         delete(path, rateLimiter);
     }
 
+    /**
+     * Recursively delete the content of the directory, but not the directory itself.
+     */
+    public static void deleteContent(Path path)
+    {
+        deleteContent(new File(path));
+    }
+
+    /**
+     * Recursively delete the content of the directory, but not the directory itself.
+     */
+    public static void deleteContent(File dir)
+    {
+        if (!dir.exists())
+            return;
+
+        for (File child : dir.tryList())
+            deleteRecursive(child.toPath());
+    }
+
+    public static List<Path> listPaths(Path path)
+    {
+        return listPaths(path, p -> true);
+    }
+
+    /**
+     * List paths in this directory that match the filter
+     */
+    public static List<Path> listPaths(Path directory, Predicate<Path> filter)
+    {
+        try (Stream<Path> stream = Files.list(directory))
+        {
+            return consistentDirectoryListings ? stream.sorted().filter(filter).collect(Collectors.toList()) 
+                                               : stream.filter(filter).collect(Collectors.toList());
```

Comment:
I suggest to move duplicated code outside the conditional operator:
```suggestion
            return (consistentDirectoryListings ? stream.sorted() 
                                               : stream).filter(filter).collect(Collectors.toList());
```
See method `list` at the beginning of this file.